invoicing/ts-classes/Line.ts

```diff
@@ -0,0 +1,12 @@
+import {YLine} from "invoicing";
+import {YTmoney} from "yupana/YClass";
+
+export class Line extends YLine {
+    cGetTotal(): number {
+        const price = this.product.price as YTmoney;
```

Comment:
This gives me 
> Error:(6, 36) TS2339: Property 'price' does not exist on type 'YTref<Product>'.

Reply:
which is true!
i can change the generation to produce a property of the class held inside the `YTref` instead of `YTref` — then the `product` will be a reference property of type `Product` instead of `YTref`...

that'll get the right types... we can think about how to achieve that later...

Reply:
i've pushed a change to the tools to "unwrap" the `TYref`... won't solve the other problem with `YTmoney` though

Reply:
The slot ```product``` should be a YTref<YProduct>```, since usually we don't have more data (we don't have the full object). Could have a method like ```getObject``` to get the refered object ?

Reply:
Well, it was 
```
    var price = line.product.price;
    var amount = line.amount
    var total = price.amount * amount;
```
in JS version. Ideally if we keep it equally simple. 
I reckon  typecasting  `<YTMoney>` or `as YTmoney ` is acceptable, but if we need to brake it to 

```
        const product = this.product as Product;
        const price = product.price as YTmoney;
```
it might be seen as counterproductive for longer chains.

Reply:
we can generate `number` properties instead of `YTinteger`, `YTdecimal` and `YTfloat`... i think that's what happens in javascript, those types are more indicators (to sql queries?) than actual runtime types of properties it seems...

Reply:
with [PR #1118](https://github.com/YupanaInc/Yupana-Framework/pull/1118)...
```ts
const price = this.product.getObject().price;
const amount = this.amount;

const total = price.amount * amount;
return total;
```
not that ugly... :)

Reply:
yeah, considering `var price = line.product.price;` never worked anyway =)